core/src/event_id.rs

```diff
@@ -65,12 +111,142 @@ impl PartialOrd for EventId {
 
 impl Ord for EventId {
     fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-        self.0.inner().cmp(&other.0.inner())
+        self.id.inner().cmp(&other.id.inner())
+    }
+}
+
+impl core::hash::Hash for EventId {
+    fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+        // Hash only the ID, not the name
+        self.id.as_int().hash(state);
     }
 }
 
 impl Display for EventId {
     fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
-        core::fmt::Display::fmt(&self.0, f)
+        match &self.name {
+            Some(name) => write!(f, "{name}"),
+            None => write!(f, "unknown event"),
+        }
+    }
+}
```

Comment:
Nit: I'm not sure if it's a good idea to implement `Display` when a useful implementation only exists for some of the instances. I think I would implement `as_str(&self) -> Option<&str>` instead, giving callers the ability to differentiate the "has name" and "has no name" cases.